back_office/invoices/models.py

```diff
@@ -1,3 +1,20 @@
 from django.db import models
+import random
+from back_office.products.models import Product
+from back_office.products.models import Discount as order_discount
+from back_office.customers.models import UserDetail
 
 # Create your models here.
+
+
+class InvoiceDetail(models.Model):
+    invoice_number = models.PositiveIntegerField(default=random.randint(0,9999))
+    customer = models.OneToOneField(UserDetail, on_delete=models.SET_NULL, null=True)
+    list_of_products = models.ManyToManyField(Product, blank=True)
+    order_discount = models.OneToOneField(order_discount, on_delete=models.SET_NULL, null=True, blank= True)
+    total_price = models.PositiveIntegerField(null=True, blank=True)
+    
+    def __str__(self):
```

Comment:
get  error: __str__ returned non-string (type int)

should be:
def __str__(self):
        return f"Invoice {self.invoice_number}"

Reply:
Thanks! I tried to fix it but got an error while doing migrations. 
_It is impossible to add a non-nullable field 'status' to order without specifying a default. This is because the database needs something to populate existing rows.
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit and manually define a default value in models.py._
 
 I don't understand where this 'status' field is.

Reply:
should work
 python manage.py makemigrations <appname> --empty